client/consul/consul.go

```diff
@@ -72,7 +72,7 @@ func NewConsulClient(config *config.ConsulConfig, logger hclog.Logger) (Client,
 		return nil, fmt.Errorf("nil consul config")
 	}
 
-	logger = logger.Named("consul")
+	logger = logger.Named("consul").With("name", config.Name)
```

Comment:
I'm not sure if there's another place in Consul where it would be worth adding this, but this struct seemed to be doing most of the logs.